### ISSUE TITLE
Linear probing and fine-tuning

### ISSUE THREAD

**Issue by mengli11235** [Navidfoumani/EEG2Rep#6]
Hi, for the "Downstream Task" section in "Rep_Learning" function from [running.py](https://github.com/Navidfoumani/EEG2Rep/blob/main/running.py#L61), line 61 to 76 is to get the results in paper section 4.3 linear probing and the rest to get the results in section 4.4 fine-tuning. Is this correct?

Meanwhile, could you disclose the 5 seeds you chose? There appears to be quite huge difference in performance when selecting different random seeds, which is caused by selecting the best model based on the validation loss for downstream tasks. This is fine for unsupervised pretraining, but not sure if it is okay for supervised training. For some seeds, the model with the lowest validation loss could be at epoch 1 or 2, which means the selected model is not really 'fine-tuned'.

**Comment by mengli11235** [comment]
Semantic_Subsequence_Preserving used the python random library. More stable performance can be achieved if using torch.randint instead.
